src/openrct2/config/Config.h

```diff
@@ -42,7 +42,7 @@ struct GeneralConfiguration
     bool UncapFPS;
     bool UseVSync;
     bool ShowFPS;
-    bool MultiThreading;
+    std::atomic_uint8_t MultiThreading;
```

Comment:
Why not just
```suggestion
    std::atomic<bool> MultiThreading;
```
?

Reply:
Bool doesn't support the toggle used here:

https://github.com/OpenRCT2/OpenRCT2/blob/339a572d03df5f0bfe5a71982c26f30606bbd762/src/openrct2-ui/windows/Options.cpp#L707

Reply:
Would it work if you replaced 1 with true?

Reply:
I tried. The issue is that there's no atomic implementation of ^= for bool, for whatever reason. There is such an implementation for integers however.

Ultimately, using atomic here probably isn't really necessary as reading and writing are separated quite a lot in the code, and they have to be for things to work anyway as the value gets saved in the lock type I introduced. But it is a 'best practice' thing as far as I know.

Reply:
This SO post talks about it and it was also modified quite recently:
https://stackoverflow.com/a/9806613

Reply:
Hm I see, it probably doesn't need to be atomic, its not going to change the value while rendering.